sockeye/inference.py

```diff
@@ -125,6 +136,10 @@ def initialize(self, max_input_length: int, get_max_output_length_function: Call
         self.encoder_module.init_params(arg_params=self.params, allow_missing=False)
         self.decoder_module.init_params(arg_params=self.params, allow_missing=False)
 
+        if self.cache_output_layer_w_b:
+            self.output_layer_w = self.params[self.decoder.output_layer.w.name].asnumpy()
```

Comment:
see my comment below in inference.py about mx.nd.take. This might save you the asnumpy here.